src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -92,7 +100,28 @@ public boolean addAll(final Collection<? extends T> collection) {
     @Override
     public boolean addAll(final int index,
             final Collection<? extends T> collection) {
-        throw new UnsupportedOperationException();
+        this.checkCollectionNotNull(collection);
+        this.checkIndexToAdd(index);
+        int startIndex = index;
+        int collectionSize = collection.size();
+
+        while (this.currentIndex + collectionSize > this.data.length - 1) {
+            this.increaseSize();
+        }
+
+        for (int j = index; j < this.currentIndex; j++) {
+            this.data[j + collectionSize] = data[j];
```

Comment:
I think here you have a bug

If u have an array has 100 elements and collection has 5 elements. Index = 0.

I think this example won't work